### PR TITLE
Add automatic publication of web components & a way to use them in an iframe

### PR DESCRIPTION
**Pull request by jahow**
This PR adds a workflow that will build the `gn-wc.js` file on every commit on master and push it on the `wc-dist` branch.

Then this file can be accessed using the following URL: https://cdn.jsdelivr.net/gh/geonetwork/geonetwork-ui@wc-dist/gn-wc.js

A new file called `wc-embedder.html` was also added in the `tools/webcomponent` directory. This is a simple HTML page
that will create a web component dynamically based on the query params it is given, thus allowing easy use of gn-ui web components in iframes.

Example: `wc-embedder.html?e=gn-dataset-preview&a=api-url=https://dev.geo2france.fr/geonetwork/srv/api`

Also fixed an issue with github-bot comments being duplicated instead of modified on pull requests.

### CODE REVIEW

File: tools/webcomponent/REAME.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+## HTML embedder
+
+The file [`wc-embedder.html`](wc-embedder.html) can be used to wrap a geonetwork-ui Web Component into a full HTML page,
+for example to be used in an [iframe](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe).
+
+To use it, specify the name and attributes of the Web Component to be created when accessing the page:
+
+```
+wc-embedder.html?e=gn-dataset-preview&a=api-url=https://dev.geo2france.fr/geonetwork/srv/api&a=primary-color=%230f4395&a=secondary-color=%238bc832&a=main-color=%23555&a=background-color=%23fdfbff
```
Comment (fgravin) [inline tools/webcomponent/REAME.md:11]: we will need a kind of api doc (up to date) for the web components...

File: .github/workflows/webcomponents.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          file: wc-dist/gn-wc.js
+          tag: ${{ github.ref }}
+          overwrite: true
+
+      - name: Publish web component to dist branch
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          force_orphan: true
```
Comment (fgravin) [inline .github/workflows/webcomponents.yml:62]: does it reset the history ?

File: .github/workflows/webcomponents.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
+name: Artifacts
+run-name: 📦 Generate artifacts for ${{ github.event_name == 'issue_comment' && 'PR' || (github.event_name == 'release' && '🏷' || '🌱') }} ${{github.event_name == 'issue_comment' && github.event.issue.number || github.ref_name}}
+
+# This workflow runs whenever a commit is pushed on main or a release is published
+on:
+  push:
+    branches:
+      - main
+  release:
+    types: [published]
```
Comment (fgravin) [inline .github/workflows/webcomponents.yml:10]: is it for each release or each commit ? the PR message is confusing.

**Review by fgravin — APPROVED** [review]
All good too, thanks, can't wait to see it deployed !

File: tools/webcomponent/REAME.md (hunk 1 above)
Comment (jahow) [reply tools/webcomponent/REAME.md:11]: Sure, and all the things related to it. Hopefully we can have a proper technical documentation website soon

File: .github/workflows/webcomponents.yml (hunk 2 above)
Comment (jahow) [reply .github/workflows/webcomponents.yml:62]: yes `force_orphan` means it always pushes an orphan branch

File: .github/workflows/webcomponents.yml (hunk 3 above)
Comment (jahow) [reply .github/workflows/webcomponents.yml:10]: Right, it's not mentioned in the PR but the workflow also runs when releases are created; then the `gn-wc.js` file is simply attached to it

**Comment by jahow** [comment]
thanks for the review!

**Pull request merged**